accounts/abi/bind/backends/simulated.go

```diff
@@ -367,7 +374,14 @@ func (b *SimulatedBackend) PendingCallContract(ctx context.Context, call ethereu
 	if err != nil {
 		return nil, err
 	}
-	return res.Return(), nil
+	// If the result contains a revert reason, try to unpack and return it.
+	if len(res.Revert()) > 0 {
+		reason, err := abi.UnpackRevert(res.Revert())
+		if err == nil {
+			return nil, fmt.Errorf("execution reverted: %v", reason)
+		}
+	}
+	return res.Return(), res.Err
```

Comment:
What if the execution is reverted with e.g. `revert(0,0)`, so that the execution _did_ revert, but `len(res.Revert())` is `0`?

Reply:
And the same, I guess, if the revert-data is non-abi-encoded

Reply:
If revert(0,0) or the revert data is non-abi encoded, the error from res.Err is returned.
This is "execution reverted" for revert(0,0). I've added a test for this